template/components/oiio.m4

```diff
@@ -0,0 +1,60 @@
+dnl BSD 3-Clause License
+dnl
+dnl Copyright (c) 2021, Intel Corporation
+dnl All rights reserved.
+dnl
+dnl Redistribution and use in source and binary forms, with or without
+dnl modification, are permitted provided that the following conditions are met:
+dnl
+dnl * Redistributions of source code must retain the above copyright notice, this
+dnl   list of conditions and the following disclaimer.
+dnl
+dnl * Redistributions in binary form must reproduce the above copyright notice,
+dnl   this list of conditions and the following disclaimer in the documentation
+dnl   and/or other materials provided with the distribution.
+dnl
+dnl * Neither the name of the copyright holder nor the names of its
+dnl   contributors may be used to endorse or promote products derived from
+dnl   this software without specific prior written permission.
+dnl
+dnl THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+dnl AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+dnl IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+dnl DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+dnl FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+dnl DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+dnl SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+dnl CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+dnl OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+dnl OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+dnl
+include(begin.m4)
+include(openexr.m4)
+
+DECLARE(`OIIO_VER',5daa9a1)
+
+ifelse(OS_NAME,ubuntu,`
+define(`OIIO_BUILD_DEPS',`ca-certificates g++ make cmake git')
+')
+
+ifelse(OS_NAME,centos,`
+define(`OIIO_BUILD_DEPS',`gcc-c++ make ifdef(`BUILD_CMAKE',,cmake) git')
+')
+
+define(`BUILD_OIIO',`
+# build OpenImageIO
+
+ARG OIIO_REPO=https://github.com/OpenImageIO/oiio.git
+RUN git clone ${OIIO_REPO} && \
+    mkdir oiio/build && \
+    cd oiio/build && \
+    git checkout OIIO_VER && \
```

Comment:
git clone and git checkout can be simplified with git clone -b VER --depth 1 ${repo}

Reply:
@xwu2git -b doesn't work for a specific commit. but if move to a specic branch/tag, source compiling will fail due to dependency changed.